it/controllers/CalculationControllerISpec.scala

```diff
@@ -221,14 +221,111 @@ class CalculationControllerISpec extends ComponentSpecBase with FeatureSwitching
         )
       }
 
-      "retrieving a calculation failed" in {
+      "retrieving a calculation failed with NOT_FOUND" in {
         enable(TaxYearOverviewUpdate)
         Given("Business details returns a successful response back")
         IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, singleBusinessResponse)
 
         And("A calculation call for 2017-18 fails")
         IndividualCalculationStub.stubGetCalculationListNotFound(testNino, "2017-18")
 
+        And("previous obligations returns a success")
+        IncomeTaxViewChangeStub.stubGetPreviousObligations(testNino,
+          LocalDate.of(2017, 4, 6),
+          LocalDate.of(2018, 4, 5),
+          ObligationsModel(Seq(
+            ReportDeadlinesModel(
+              "ABC123456789",
+              List(ReportDeadlineModel(
+                LocalDate.of(2017, 12, 28),
+                LocalDate.of(2018, 2, 3),
+                LocalDate.of(2018, 2,4),
+                "Quarterly",
+                Some(LocalDate.of(2018, 2, 2)),
+                "#001"
+              ))
+            )
+          ))
+        )
+
+        And("current obligations returns a success")
+        IncomeTaxViewChangeStub.stubGetReportDeadlines(testNino,
+          ObligationsModel(Seq(
+            ReportDeadlinesModel(
+              "ABC123456789",
+              List(ReportDeadlineModel(
+                LocalDate.of(2017, 11, 28),
+                LocalDate.of(2018, 1, 3),
+                LocalDate.of(2018, 1,4),
+                "Quarterly",
+                Some(LocalDate.of(2018, 1, 2)),
+                "#001"
+              ))
+            )
+          ))
+        )
+
+        When(s"I call GET ${controllers.routes.CalculationController.renderTaxYearOverviewPage(testYearInt).url}")
+        val res = IncomeTaxViewChangeFrontend.getCalculation(testYear)
+
+        Then("I check all calls expected were made")
+        verifyIncomeSourceDetailsCall(testMtditid)
+        IndividualCalculationStub.verifyGetCalculationList(testNino, "2017-18")
+
+        And("Internal server error is returned")
```

Comment:
Optional suggestion: make the description matching the test case, something like:
```Scala
 And("The expected result is returned")
```

Reply:
Done